File: example/c/lint-control-flow.c
#include <stdio.h>

int main(void) {
  int values[] = {10, 20, 30, 40, 50};
  int total = 0;

  // 配列は5件なのに4件しか処理しない。
  for (int i = 0; i < 4; i++) {
    total += values[i];
  }

  int ready = 0;
  // 比較ではなく代入になっている。
  if (ready = 1) {
    puts("ready");
  }

  int retries = 3;
  // 条件が代入なのでループに入らない。
  while (retries = 0) {
    retries--;
  }

  // case 0にbreakがなく、case 1へフォールスルーする。
  switch (total) {
    case 0:
      puts("empty");
    case 1:
      puts("one");
      break;
    default:
      break;
  }

  int matrix[2][3] = {
    {1, 2, 3},
    {4, 5, 6},
  };
  int matrix_total = 0;
  // 2行あるのに先頭1行しか処理しない。
  for (int row = 0; row < 1; row++) {
    for (int column = 0; column < 3; column++) {
      matrix_total += matrix[row][column];
    }
  }

  printf("%d %d\n", total, matrix_total);
  return 0;
}

File: example/javascript/lint-control-flow.js
const values = [10, 20, 30, 40, 50]
let total = 0

// 配列は5件なのに4件しか処理しない。
for (let i = 0; i < 4; i++) {
  total += values[i]
}

let ready = false
// 比較ではなく代入になっている。
if (ready = true) {
  console.log('ready')
}

let retries = 3
// whileの条件も比較ではなく代入になっている。
while (retries = 0) {
  retries--
}

const mode = 'sum'
switch (mode) {
  case 'sum':
    break
    console.log('break後なので到達できない')
  case 'average':
    console.log('average')
    break
  default:
    console.log('unknown')
}

const matrix = [
  [1, 2, 3],
  [4, 5, 6],
]

// 二次元配列は2行あるのに先頭1行しか処理しない。
for (let row = 0; row < 1; row++) {
  for (let column = 0; column < matrix[row].length; column++) {
    total += matrix[row][column]
  }
}

// breakのタイプミス。未定義の識別子として検出される。
for (let i = 0; i < values.length; i++) {
  if (values[i] === 30) brake
}

console.log(total)
